compiler/doc/rng.md

```diff
@@ -0,0 +1,12 @@
+# ByteIR RNG CustomCall Converter
+
+## RNG Uniform Converter
+
+We use Philox4×32-10 and Normalization to implement Uniform Distribution Generator for floating-point value. Philox4×32-10 is a counter-based random number generator that takes a 4 32-bit integer and 2 32-bit keys as input and produces 4 32-bit uniformly distributed random **integers** as output. So using Philox4×32-10 we can get random integers, and then normalization will be taken to convert random integers to floating-point value.
+It’s noteworthy that in order to ensure that it can be fused with other elementwise ops, for each Philox4×32-10, only the first output will be taken, and the remaining three outputs will be discarded. This could potentially be optimized to enhance performance
+
+## RNG Normal Converter
+
+We use Box–Muller transform and Philox4×32-10 to implemet Normal Distribution Generator. Box–Muller transform takes two samples from the uniform distribution on the interval (0,1) and maps them to two standard, normally distributed samples.
+So We use the Philox4×32-10 and Normalization to generate two uniformly distributed random numbers whose range from 0 to 1 (just like the Uniform Distribution Generator, but this time we need to take two of the four Philox4×32-10 outputs and discard the remaining two) and then use Box–Muller transform to map them to standard, normally distributed values.
+So to generate a normal distributed random numbers, We need to perform a philox4×32-10 and an Box-Muller transform once. But if we can allow multiple random numbers to be generated at once, since a philox4×32-10 can generate 4 uniformly distributed random numbers, we can generate 6 pairs of inputs to the Box-Muller transform and generate 12 normal distributed random numbers with one philox4×32-10. It can greatly reduce the number of executions of philox4×32-10.
```

Comment:
Could you add `hlo-to-linalg` lowering case to show details? Also add math descriptions?

Reply:
And add future work which could provide optimization opportunites.

Reply:
Sure!